pytorch_lightning/trainer/trainer.py

```diff
@@ -1091,6 +1091,7 @@ def _run(
         if hasattr(model, "hparams"):
             parsing.clean_namespace(model.hparams)
 
+        self._callback_connector._attach_model_callbacks(model)
```

Comment:
why before ""verify_loop_configurations" and not under `attach_model_logging_functions`? That way you don't need to pass the pl_module reference as we want to remove the argument

cc @daniellepintz

Reply:
under `verify_loop_configurations` we have certain checks for callback hooks. So here now we make sure we check them for callbacks returned from `LightningModule` as well.

Reply:
Pushed a change reordering so the argument change is not necessary.